M118 | Homing | ABL | MQTT

nicedevil007 opened the pull request:
I added some features to get the printer faster to the point where it is ready to print, especially if you print multiple prints after an other. No Homing is needed if the printer was already homed after the last time the steppers were turned off f.e.

Same for Z_TILT_ADJUST.

M118 is shown at the top of Klipper Screen, M117 not

Nozzle Cleaning Option before the print starts

MQTT => for all guys out there that use MQTT and configured it we can send messages with my additions to the broker and do what ever we want with it. F.e. let Alexa tell us that our print is done or filament runout is detected... or just a simple pushmessage via. pushover or something like this.

miklschmidt commented:
Thank you! Unfortunately there's a bit too much stuff here. Not sure MQTT should be a default RatOS thing. Furthermore nozzle cleaning is unnecessary with the prime macros. I've added conditional homing where it makes sense and made sure all M117's are followed by a RESPOND. Thanks for you contribution :)

nicedevil007 commented:
> Thank you! Unfortunately there's a bit too much stuff here. Not sure MQTT should be a default RatOS thing. Furthermore nozzle cleaning is unnecessary with the prime macros. I've added conditional homing where it makes sense and made sure all M117's are followed by a RESPOND. Thanks for you contribution :)

Yeah and I know that it was a bit confusing because of my knownledge about the klipper programming isn't the best right now, still on learning.

Maybe a good solution (I found out a few days before on the official klipper docu) would something like this:

```
[gcode_macro M117]
rename_existing: M117.1
gcode:
  M117.1 { rawparams }
  M118 { rawparams }
```

Then we only need M117 in macros and then both is shown... M117 as normal and M118 in the console (dont forget the respond thing for this).

Here an example what I added to SuperSlicer Layerchange GCODE:

`M117 Layer {layer_num +1}/[total_layer_count], z = [layer_z]`

Output looks like this:
![image](https://user-images.githubusercontent.com/17103076/159138207-ef5dd8b2-5b4b-4cd6-8455-d36083700aa6.png)

And this:
![image](https://user-images.githubusercontent.com/17103076/159138214-f8ad0a63-7df1-4acb-abb7-c6bb45de88e7.png)

MQTT was intended to be as optional thats why I made an option in your RatOS variables.
